runtime/battery-station/src/lib.rs

```diff
@@ -101,32 +112,88 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     state_version: 1,
 };
 
+#[derive(scale_info::TypeInfo)]
+pub struct ContractsCallfilter;
+
+impl Contains<RuntimeCall> for ContractsCallfilter {
+    fn contains(runtime_call: &RuntimeCall) -> bool {
+        #[allow(clippy::match_like_matches_macro)]
+        match runtime_call {
+            RuntimeCall::AssetManager(transfer { .. }) => true,
+            RuntimeCall::PredictionMarkets(inner_call) => {
+                match inner_call {
+                    buy_complete_set { .. } => true,
+                    deploy_swap_pool_and_additional_liquidity { .. } => true,
+                    deploy_swap_pool_for_market { .. } => true,
+                    dispute { .. } => true,
+                    // Only allow CPMM markets using Authorized or SimpleDisputes dispute mechanism
+                    create_market {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        scoring_rule: ScoringRule::CPMM,
+                        ..
+                    } => true,
+                    create_cpmm_market_and_deploy_assets {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        ..
+                    } => true,
+                    edit_market {
+                        dispute_mechanism:
+                            MarketDisputeMechanism::Authorized | MarketDisputeMechanism::SimpleDisputes,
+                        scoring_rule: ScoringRule::CPMM,
+                        ..
+                    } => true,
```

Comment:
Are SimpleDisputes really necessary here? And especially I would disallow global disputes, until it's properly implemented alongside court. Specifically because of [this](https://github.com/zeitgeistpm/zeitgeist/pull/912#discussion_r1102490562). So rather not allow the additional possibility of smart contracts to break my assumption there (that nobody will create a global dispute on battery station).

Reply:
If you had allowed global disputes, you would have also included the GlobalDispute pallet and its calls. But since I would like to remove it from the list here, we are fine.

Reply:
[GlobalDisputes](https://github.com/zeitgeistpm/zeitgeist/blob/ea97e10b5f96b47f5847c14d436196c0717ebb11/runtime/battery-station/src/lib.rs#L141) and SimpleDisputes were already enabled on Battery Station, that's why I excluded the calls from the smart contract call filter. I am fine to leave them out though.